builtin/logical/pki/crl_util.go

```diff
@@ -1550,6 +1767,87 @@ func getRevokedCertEntries(sc *storageContext, issuerIDCertMap map[issuerID]*x50
 	return unassignedCerts, revokedCertsMap, nil
 }
 
+func getUnifiedRevokedCertEntries(sc *storageContext, issuerIDCertMap map[issuerID]*x509.Certificate, isDelta bool) ([]pkix.RevokedCertificate, map[issuerID][]pkix.RevokedCertificate, error) {
+	// Getting unified revocation entries is a bit different than getting
+	// the local ones. In particular, the full copy of the certificate is
+	// unavailable, so we'll be able to avoid parsing the stored certificate,
+	// at the expense of potentially having incorrect issuer mappings.
+	var unassignedCerts []pkix.RevokedCertificate
+	revokedCertsMap := make(map[issuerID][]pkix.RevokedCertificate)
+
+	listingPath := unifiedRevocationReadPathPrefix
+	if isDelta {
+		listingPath = unifiedDeltaWALPrefix
+	}
+
+	// First, we find all clusters that have written certificates.
+	clusterIds, err := sc.Storage.List(sc.Context, listingPath)
```

Comment:
This doesn't work for two reasons:

1. We need to be relative to `listingPath`, and not to `unifiedRevocationReadPathPrefix`, in particular, to support listing Delta WAL entries.
2. We don't need the path prepended, or rather, we want it optionally with the path or not (so that we can read the full revocation entry when we see a serial under the delta WAL -- we take that delta WAL's serial + cluster and put it on the `unifiedRevocationReadPathPrefix`. 

But yes, I was eventually thinking about refactoring the `getUnifiedRevocationBySerial` to take an optional cluster hint that'd be checked first, and then also perhaps take a prefix.

Reply:
Ah yes I see it now, was reading it too quickly.